### PR TITLE
feat: [v6] new component Masonry

### LINKED ISSUE #40910: New Component Masonry
https://mui.com/material-ui/react-masonry/

### LINKED ISSUE #48842: Masonary Component Required
### What problem does this feature solve?

I was looking for Masonry Component just like  Material UI (https://mui.com/material-ui/react-masonry/)  . Although It can be done with Flex and Grid but it will ease the coding if we have masonry just like Pinterest use and we have Material UI as an example for Masonry

### What does the proposed API look like?

Props 
1. children : The content of the component.
2. columns : Number of columns.
3. spacing:  Defines the space between children.
4. sequential:  Allows using sequential order rather than adding to shortest column

### PR DESCRIPTION
**Pull request by OysterD3**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [x] 🆕 New feature
- [ ] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

> - Describe the source of related requirements, such as links to relevant issue discussions.
> - For example: close #xxxx, fix #xxxx

- close #48842 
- close #40910 

### 💡 Background and Solution

> - The specific problem to be addressed.
> - List the final API implementation and usage if needed.
> - If there are UI/interaction changes, consider providing screenshots or GIFs.

#51659 

### 📝 Change Log

> - Read [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) like a cat tracks a laser pointer.
> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Add new Masonry component |
| 🇨🇳 Chinese | 新增瀑布流布局 |

### CODE REVIEW

**Comment by zombieJ** [comment]
https://github.com/ant-design/ant-design/discussions/51659  要先过个 RFC

File: components/masonry/style/index.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      [`& > ${componentCls}-item`]: {
+        boxSizing: 'border-box',
+        transform: `translate(var(--${prefixCls}-item-translate-x), var(--${prefixCls}-item-translate-y))`,
+        insetInlineStart: 0,
+        top: 0,
+        position: 'absolute',
+        width: `var(--${prefixCls}-item-width)`,
+        height: `var(--${prefixCls}-item-height)`,
+        overflow: 'hidden',
+        transition: 'transform 0.1s ease-out',
```
Comment (li-jia-nan) [inline components/masonry/style/index.ts:31]: 这个 0.1 换成 `motionDurationFast`

**Comment by li-jia-nan** [comment]
这个需要提交到 next 分支

**Comment by OysterD3** [comment]
#52162

**Pull request closed without merging**